docarray/array/queryset/lookup.py

```diff
@@ -0,0 +1,248 @@
+"""
+
+Originally from https://github.com/naiquevin/lookupy
+
+The library is provided as-is under the MIT License
+
+Copyright (c) 2013 Vineet Naik (naikvin@gmail.com)
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ... import Document
+import re
+from functools import partial
+
+
+def lookup(key, val, doc: 'Document') -> bool:
+    """Checks if key-val pair exists in doc using various lookup types
+
+    The lookup types are derived from the `key` and then used to check
+    if the lookup holds true for the document::
+
+        >>> lookup('text__exact', 'hello', doc)
+
+    The above will return True if doc.text == 'hello' else False
+
+    :param key  : the field name to find
+    :param val  : object to match the value in the document against
+    :param doc : the document to match
+    """
+    get_key, last = dunder_partition(key)
+    value = doc._get_attributes(get_key)
```

Comment:
```suggestion
    get_key, last = dunder_partition(key)
    value = doc._get_attributes(get_key)

    regex = re.compile(r'\$\{\{\s*([a-zA-Z0-9_]*)\s*}}')
    m = regex.match(val)
    if isinstance(val, str) and m:
        val_key = regex.sub(lambda matchobj: matchobj.group(1), val)
        val = doc._get_attributes(val_key)
```

should allow this syntax:
`da._filter({'tags__a': {'$gte': '${{tags__c}}'}})`

Reply:
Like this idea, and will add it soon

Reply:
Instead use `${{...}}` syntax, I choose to use pythonic-style placeholder syntax  `{...}` 
```
da._filter({'tags__a': {'$gte': '{tags__c}'}})
```